### PR TITLE
CLDR-15405 Add initialPatterns, sampleNames & data; PathHeader functions for ordering

### PR DESCRIPTION
**Pull request by pedberg-icu**
CLDR-15405

- [ ] This PR completes the ticket.

Finish up the remaining changes to dtd, coverage, PathHeader etc. modulo discussion about changing some of the added items, and add new data items for root/English:
- Add initialPattern element with type=initial|initialSequence, and data for root/English. Since this is PCDATA, it is also required to have an alt attribute, so it allows alt=variant.
- Add sampleName element with item=minimal|simple|full|multiword, and child element nameField with type=prefix|given|given2|surname|surname2|suffix, and initial data for English. Since nameField is PCDATA, it also allows alt=variant. **Note: we need to discuss (in meeting): (1) whether "item" is the best name for the sampleName attribute; (2) what additional values might be needed besides minimal|simple|full|multiword; and (3) whether the initial English data is what we want.**
- Fix the namePattern alt attribute to be 1|2, not 2|3.
- Update coverage, PathHeader, PathDescription, TextExampleGenerator, etc. for the above changes.
- Add PathHeader ordering functions personNameOrder and sampleNameOrder, use them in  PathHeader.txt for correct ordering of paths.

### CODE REVIEW

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/PathDescription.java
Comment (macchiati) [inline tools/cldr-code/src/main/java/org/unicode/cldr/util/PathDescription.java:382]: I think we also need to do the PlaceHolderDescriptions (but not in this PR). I think our tests might ignore placeholders that aren't {0}, {1}, etc.

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/PathHeader.java
Comment (macchiati) [inline tools/cldr-code/src/main/java/org/unicode/cldr/util/PathHeader.java:1868]: I think this needs to be refined.
We might get collisions. 
long = 0 would be zero, and collide with addressing = 0.
Also medium + short (1 + 2) = monogram (3).

Comment (macchiati) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/PathHeader.java:1868]: What I suggest instead is having one List with all the values. When you get an index, add (1 << index) to the order.

**Review by macchiati — CHANGES_REQUESTED** [review]
Changes could go into a later PR, if you want to get this progress in now.

File: common/main/en.xml
Comment (richgillam) [inline common/main/en.xml:9258]: For a Spanish name like this, would you want a `surname2` field as well?  Or would an example like that go somewhere else?

**Review by richgillam — COMMENTED** [review]
I don't feel qualified to review a lot of this; I don't know enough about what it's doing or how it's supposed to work.  I'll defer to Mark.

File: common/main/en.xml
Comment (pedberg-icu) [reply common/main/en.xml:9258]: We need to discuss in the meeting what kinds of example names we want for English.

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/PathDescription.java
Comment (pedberg-icu) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/PathDescription.java:382]: I will add that to the TODOs in the agenda

File: common/main/en.xml
Comment (pedberg-icu) [reply common/main/en.xml:9268]: Since thee were so many names in given2 I just moved the first to add to given.

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/PathHeader.java
Comment (pedberg-icu) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/PathHeader.java:1868]: Oh, good idea, I will do that; I forgot we could have multiple instances of the same type.

**Comment by pedberg-icu** [comment]
@macchiati I made some changes to address your feedback, thanks. I will leave un-squashed for now so you can just review the deltas, then will squash (and will need a re-approval).

File: common/main/root.xml
Comment (macchiati) [inline common/main/root.xml:5373]: I think more languages use spaces between initials than not, so maybe this should also be space separated.

File: common/main/en.xml
Comment (macchiati) [reply common/main/en.xml]: ok

File: tools/cldr-code/src/main/java/org/unicode/cldr/util/PathDescription.java
Comment (macchiati) [reply tools/cldr-code/src/main/java/org/unicode/cldr/util/PathDescription.java:382]: SGTM

File: common/main/en.xml
Comment (macchiati) [reply common/main/en.xml:9258]: We don't have to have names for specific people. I think it is actually useful to have someone with surname2 for English, to show that we *don't* included in normal formatting there.

**Comment by macchiati** [comment]
> @macchiati I made some changes to address your feedback, thanks. I will leave un-squashed for now so you can just review the deltas, then will squash (and will need a re-approval).

Ok, will do now.

**Review by macchiati — APPROVED** [review]
Changes you made look good. If you want to merge, you can address other comments in a later PR.

File: common/main/root.xml
Comment (pedberg-icu) [reply common/main/root.xml]: @macchiati OK, I added spaces (I only had it without spaces because that is the way it was in the agenda to-dos), and have squashed. Those are the only changes, needs re-approval to merge, thanks.

File: common/main/en.xml
Comment (pedberg-icu) [reply common/main/en.xml:9258]: Sounds good. I need to update the names here based on discussion in meeting; also need to add more same name item types. Will do that all in another PR.

**Comment by pedberg-icu** [comment]
@macchiati Needs re-approval after squashing (and adding spaces in the root initialSequence pattern), thanks. - Peter

**Review by macchiati — APPROVED** [review]

**Comment by macchiati** [comment]
Now mergeable

**Pull request merged**
